### ISSUE TITLE
available-versions for java invalid

### ISSUE THREAD

**Issue by hohwille** [devonfw/ide#933]
### Expected behavior

As a devonfw-ide user, I want to make use of the new version support including completion, etc. so that my life gets easier maintaining versions. Therefore available-versions should contain only valid stable versions of the according tool that are available.

### Actual behavior

Currently [available-versions for java](https://github.com/devonfw/ide-mirrors/blob/master/java/available-versions) contains `19_36` as latest version, so I tried to install it:

```
devon java set version 19_36
devon java setup
Updating java from version 11.0.13_8 to version 19_36...
Starting installation of java in version 19_36 to /d/projects/devonfw/software/java
/d/projects/devonfw/scripts/command/../functions: line 398:
******** ATTENTION ********
None of the followin path(s) exists:
/d/projects/devonfw/mirrors/java/Temurin/os-mappings
/d/projects/devonfw/mirrors/java/os-mappings
/os-mappings
We are sorry for the inconvenience. Please check the above errors, resolve them and try again.: No such file or directory
Trying to download java-19_36-windows.zip from https://github.com/adoptium/temurin19-binaries/releases/download/jdk-11/OpenJDK19U-jdk_x64_windows_hotspot_19_36.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404
Failed to download https://github.com/adoptium/temurin19-binaries/releases/download/jdk-11/OpenJDK19U-jdk_x64_windows_hotspot_19_36.zip with exit code 22

******** ATTENTION ********
Download of java in version 19_36 failed after trying 1 mirror(s).
We are sorry for the inconvenience. Please check the above errors, resolve them and try again.
```

### Steps to reproduce (bug) / Use Case of feature request (enhancement)

1. `devon java set version 19_36`
2. `devon java setup`

### Related/Dependent Issues

* #826
* #931

### Comments/Hints:

There is a typo in the error message that I already fixed (`followin` is missing a `g`).
Besides if `os-mapping` is required, we need to add it for `java`.
Further, the last entry `/os-mappings` looks odd and seems to indicate another bug.

### Affected version:

* WIP of 2022.08.004

**Comment by hohwille** [comment]
When I go to https://adoptium.net I get `17.0.4.1_1` as latest version.
IMHO this is wrong: https://github.com/devonfw/ide-mirrors/blob/0a40bb51572677c4d429e3dbf1a8c97bad48bc4a/java/Temurin/versions-url

Or we have to filter `debugimage` in https://github.com/devonfw/ide-mirrors/blob/0a40bb51572677c4d429e3dbf1a8c97bad48bc4a/java/Temurin/versions-url-op

**Comment by alfeilex** [comment]
I can't reproduce the bug. Comparing the two download links, it seems that the `${code}` variable in your log is not resolved correctly.

Instead of 
https://github.com/adoptium/temurin19-binaries/releases/download/jdk-19%2B36/OpenJDK19U-jdk_x64_windows_hotspot_19_36.zip

it is 
https://github.com/adoptium/temurin19-binaries/releases/download/jdk-11/OpenJDK19U-jdk_x64_windows_hotspot_19_36.zip

My log: 
```Updating java from version 11.0.13_8 to version 19_36...
Starting installation of java in version 19_36 to /c/projects/devonfw-ide-scripts-2022.08.003/software/java
Trying to download java-19_36-windows.zip from https://github.com/adoptium/temurin19-binaries/releases/download/jdk-19%2B36/OpenJDK19U-jdk_x64_windows_hotspot_19_36.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0
100  187M  100  187M    0     0  4861k      0  0:00:39  0:00:39 --:--:-- 6855k
Download of java-19_36-windows.zip from https://github.com/adoptium/temurin19-binaries/releases/download/jdk-19%2B36/OpenJDK19U-jdk_x64_windows_hotspot_19_36.zip succeeded.
Success: run command unzip
Successfully extracted/moved archive java-19_36-windows.zip to updates/extracted
Creating backup by moving existing /c/projects/devonfw-ide-scripts-2022.08.003/software/java to /c/projects/devonfw-ide-scripts-2022.08.003/updates/backups/22-10-11
Success: run command mv
Successfully installed java
The software java has been added. You need to rerun 'devon' command without arguments or restart your terminal to update your PATH so the newly installed software will be found.
openjdk version "19" 2022-09-20
OpenJDK Runtime Environment Temurin-19+36 (build 19+36)
OpenJDK 64-Bit Server VM Temurin-19+36 (build 19+36, mixed mode, sharing)
Success: verify installation of Java ('/c/projects/devonfw-ide-scripts-2022.08.003/software/java/bin/java')
```

Am I missing something?

**Comment by hohwille** [comment]
OK, so I figured this out.
It is actually a bug of PR #934
What I missed out is this magic to determine the `code` from the `version`:
https://github.com/devonfw/ide/blob/b41a60a3025ffca93985faff937f374519284b81/scripts/src/main/resources/scripts/command/java#L55

This gives us a new challenge for #878 and #893:
We want to move the most logic to `doInstall` including to determine the "real" version in there.
As however, we have still some workarounds in commandlets to compute `code` from the "real" `version` this stops working as it is not yet known.

**Comment by hohwille** [comment]
This is fixed now in PR #934.
As the code with the bug was never merged, it does not make sense to add this bug ticket to the CHANGELOG.
I am therefore just closing it.
